Exception: Jackson jackson-core/jackson-dataformat-cbor incompatible library version detected

Running Exasol on AWS (version 7.0.3) and using the current release I get the following error message:

`
[Code: 0, SQL State: 22002]  VM error: F-UDF-CL-LIB-1127: F-UDF-CL-SL-JAVA-1002: F-UDF-CL-SL-JAVA-1013: 
com.exasol.ExaUDFException: F-UDF-CL-SL-JAVA-1080: Exception during run 
com.amazonaws.SdkClientException: Unable to marshall request to JSON: Jackson jackson-core/jackson-dataformat-cbor incompatible library version detected.
You have two possible resolutions:
		1) Ensure the com.fasterxml.jackson.core:jackson-core & com.fasterxml.jackson.dataformat:jackson-dataformat-cbor libraries on your classpath have the same version number
		2) Disable CBOR wire-protocol by passing the -Dcom.amazonaws.sdk.disableCbor property or setting the AWS_CBOR_DISABLE environment variable (warning this may affect performance)
com.amazonaws.services.kinesis.model.transform.DescribeStreamRequestProtocolMarshaller.marshall(DescribeStreamRequestProtocolMarshaller.java:59)
com.amazonaws.services.kinesis.AmazonKinesisClient.executeDescribeStream(AmazonKinesisClient.java:861)
com.amazonaws.services.kinesis.AmazonKinesisClient.describeStream(AmazonKinesisClient.java:846)
com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader$.getAllShardsFromStream(KinesisShardsMetadataReader.scala:77)
com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader$.run(KinesisShardsMetadataReader.scala:33)
com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader.run(KinesisShardsMetadataReader.scala)
com.exasol.ExaWrapper.run(ExaWrapper.java:196)
Caused by: java.lang.RuntimeException: Jackson jackson-core/jackson-dataformat-cbor incompatible library version detected.
You have two possible resolutions:
		1) Ensure the com.fasterxml.jackson.core:jackson-core & com.fasterxml.jackson.dataformat:jackson-dataformat-cbor libraries on your classpath have the same version number
		2) Disable CBOR wire-protocol by passing the -Dcom.amazonaws.sdk.disableCbor property or setting the AWS_CBOR_DISABLE environment variable (warning this may affect performance)
com.amazonaws.protocol.json.SdkCborGenerator.getBytes(SdkCborGenerator.java:69)
com.amazonaws.protocol.json.internal.JsonProtocolMarshaller.finishMarshalling(JsonProtocolMarshaller.java:190)
com.amazonaws.protocol.json.internal.NullAsEmptyBodyProtocolRequestMarshaller.finishMarshalling(NullAsEmptyBodyProtocolRequestMarshaller.java:53)
com.amazonaws.services.kinesis.model.transform.DescribeStreamRequestProtocolMarshaller.marshall(DescribeStreamRequestProtocolMarshaller.java:57)
	... 10 more
Caused by: java.lang.NoSuchMethodError: 'com.fasterxml.jackson.core.json.JsonWriteContext com.fasterxml.jackson.dataformat.cbor.CBORGenerator.getOutputContext()'
com.fasterxml.jackson.dataformat.cbor.CBORGenerator.close(CBORGenerator.java:903)
com.amazonaws.protocol.json.SdkJsonGenerator.close(SdkJsonGenerator.java:267)
com.amazonaws.protocol.json.SdkJsonGenerator.getBytes(SdkJsonGenerator.java:282)
com.amazonaws.protocol.json.SdkCborGenerator.getBytes(SdkCborGenerator.java:67)
	... 13 more
 (Session: 1684968654543585281)`


Currently I'm adding %jvmoption -Dcom.amazonaws.sdk.disableCbor=true;  to all UDFs to make it work:
`--/
CREATE OR REPLACE JAVA SET SCRIPT KINESIS_SCHEMA.KINESIS_METADATA (...) 
EMITS (KINESIS_SHARD_ID VARCHAR(130), SHARD_SEQUENCE_NUMBER VARCHAR(2000)) AS
%jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
%scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
%jar /buckets/bfsdefault/kinesis/exasol-kinesis-connector-extension-0.1.0.jar;
/
;



--/
CREATE OR REPLACE JAVA SET SCRIPT KINESIS_SCHEMA.KINESIS_IMPORT (...) EMITS (...) AS
%jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
%scriptclass com.exasol.cloudetl.kinesis.KinesisShardDataImporter;
%jar /buckets/bfsdefault/kinesis/exasol-kinesis-connector-extension-0.1.0.jar;
/
;



--/
CREATE OR REPLACE JAVA SET SCRIPT KINESIS_SCHEMA.KINESIS_CONSUMER (...) EMITS (...) AS
%jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
%scriptclass com.exasol.cloudetl.kinesis.KinesisImportQueryGenerator;
%jar /buckets/bfsdefault/kinesis/exasol-kinesis-connector-extension-0.1.0.jar;
/
;`

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,8 +15,7 @@ env:
 
 before_install:
   - git fetch --tags
-  - docker pull exasol/docker-db:latest
-  - docker pull exasol/docker-db:6.2.7-d1
+  - docker pull exasol/docker-db:7.0.4
 
 matrix:
   include:

--- a/doc/changes/changes_0.1.1.md
+++ b/doc/changes/changes_0.1.1.md
@@ -2,7 +2,7 @@
 
 Code name: 
 
-## Bug
+## Bug Fixes
 
 * #18: Fixed jackson-core/jackson-dataformat-cbor incompatible library version.
 
@@ -23,3 +23,4 @@ Code name:
 * Updated `com.exasol:exasol-testcontainers:3.0.0` to `3.4.0`
 * Updated `org.testcontainers:localstack:1.14.3` to `1.15.0`
 * Updated `com.amazonaws:aws-java-sdk-core:1.11.860` to `1.11.916`
+* Updated `com.exasol:import-export-udf-common-scala:0.1.0` to `0.2.0`

--- a/doc/changes/changes_0.1.1.md
+++ b/doc/changes/changes_0.1.1.md
@@ -1,6 +1,10 @@
 # Kinesis Connector Extension 0.1.1, released 2020-??-??
 
-## Feature
+Code name: 
+
+## Bug
+
+* #18: Fixed jackson-core/jackson-dataformat-cbor incompatible library version.
 
 ## Refactoring
 
@@ -12,3 +16,10 @@
 * #15: Added a link to the development guide.
 
 ## Dependency updates
+
+* Added `com.fasterxml.jackson.core:jackson-dataformat-cbor:2.12.0`
+* Updated `com.fasterxml.jackson.core:jackson-databind:2.11.2` to `2.12.0`
+* Updated `com.fasterxml.jackson.core:jackson-module-scala:2.11.2` to `2.12.0`
+* Updated `com.exasol:exasol-testcontainers:3.0.0` to `3.4.0`
+* Updated `org.testcontainers:localstack:1.14.3` to `1.15.0`
+* Updated `com.amazonaws:aws-java-sdk-core:1.11.860` to `1.11.916`

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -9,15 +9,15 @@ object Dependencies {
   // Runtime dependencies versions
   private val ExasolVersion = "6.1.7"
   private val ImportExportUDFVersion = "0.1.0"
-  private val AwsJavaSdkVersion = "1.11.860"
+  private val AwsJavaSdkVersion = "1.11.916"
+  private val FasterxmlJacksonVersion = "2.12.0"
 
   // Test dependencies versions
   private val ScalaTestVersion = "3.2.2"
   private val ScalaTestPlusVersion = "1.0.0-M2"
   private val MockitoCoreVersion = "3.5.10"
-  private val ExasolTestContainersVersion = "3.0.0"
-  private val TestContainersLocalstackVersion = "1.14.3"
-  private val FasterxmlJacksonVersion = "2.11.2"
+  private val ExasolTestContainersVersion = "3.4.0"
+  private val TestContainersLocalstackVersion = "1.15.0"
 
   val Resolvers: Seq[Resolver] = Seq(
     "Confluent Maven Repo" at "https://packages.confluent.io/maven/",
@@ -30,7 +30,8 @@ object Dependencies {
     "com.amazonaws" % "aws-java-sdk-core" % AwsJavaSdkVersion,
     "com.amazonaws" % "aws-java-sdk-kinesis" % AwsJavaSdkVersion,
     "com.fasterxml.jackson.core" % "jackson-databind" % FasterxmlJacksonVersion,
-    "com.fasterxml.jackson.module" %% "jackson-module-scala" % FasterxmlJacksonVersion
+    "com.fasterxml.jackson.module" %% "jackson-module-scala" % FasterxmlJacksonVersion,
+    "com.fasterxml.jackson.dataformat" % "jackson-dataformat-cbor" % FasterxmlJacksonVersion
   )
 
   lazy val TestDependencies: Seq[ModuleID] = Seq(

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -8,7 +8,7 @@ object Dependencies {
 
   // Runtime dependencies versions
   private val ExasolVersion = "6.1.7"
-  private val ImportExportUDFVersion = "0.1.0"
+  private val ImportExportUDFVersion = "0.2.0"
   private val AwsJavaSdkVersion = "1.11.916"
   private val FasterxmlJacksonVersion = "2.12.0"
 

--- a/src/it/scala/com/exasol/cloudetl/kinesis/KinesisAbstractIntegrationTest.scala
+++ b/src/it/scala/com/exasol/cloudetl/kinesis/KinesisAbstractIntegrationTest.scala
@@ -1,15 +1,16 @@
 package com.exasol.cloudetl.kinesis
 
-import java.io.File
-import java.nio.file.Paths
-import java.sql.ResultSet
-
 import com.amazonaws.SDKGlobalConfiguration.AWS_CBOR_DISABLE_SYSTEM_PROPERTY
 import com.amazonaws.services.kinesis.{AmazonKinesis, AmazonKinesisClientBuilder}
-import com.exasol.containers.{ExasolContainer, ExasolContainerConstants}
+import com.exasol.containers.ExasolContainer
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.funsuite.AnyFunSuite
 import org.testcontainers.containers.localstack.LocalStackContainer
+import org.testcontainers.utility.DockerImageName
+
+import java.io.File
+import java.nio.file.Paths
+import java.sql.ResultSet
 
 trait KinesisAbstractIntegrationTest extends AnyFunSuite with BeforeAndAfterAll {
   val JAR_DIRECTORY_PATTERN = "scala-"
@@ -18,11 +19,10 @@ trait KinesisAbstractIntegrationTest extends AnyFunSuite with BeforeAndAfterAll 
   val TEST_SCHEMA_NAME = "kinesis_schema"
   var assembledJarName: String = _
 
-  val exasolContainer = new ExasolContainer(
-    ExasolContainerConstants.EXASOL_DOCKER_IMAGE_REFERENCE
-  )
+  val exasolContainer = new ExasolContainer("7.0.4")
   val kinesisLocalStack: LocalStackContainer =
-    new LocalStackContainer().withServices(LocalStackContainer.Service.KINESIS)
+    new LocalStackContainer(DockerImageName.parse("localstack/localstack:0.12.3"))
+      .withServices(LocalStackContainer.Service.KINESIS)
 
   private[this] var connection: java.sql.Connection = _
   var statement: java.sql.Statement = _
@@ -98,7 +98,6 @@ trait KinesisAbstractIntegrationTest extends AnyFunSuite with BeforeAndAfterAll 
     statement.execute(
       s"""CREATE OR REPLACE JAVA SET SCRIPT KINESIS_METADATA (...)
          |EMITS (KINESIS_SHARD_ID VARCHAR(130), SHARD_SEQUENCE_NUMBER VARCHAR(2000)) AS
-         |     %jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
          |     %scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
          |     %jar /buckets/bfsdefault/default/$assembledJarName;
          |/

